darts/explainability/tft_explainer.py

```diff
@@ -0,0 +1,249 @@
+        model: TFTModel,
+    ):
+        """
+        Explainer class for the TFT model.
+
+        Parameters
+        ----------
+        model
+            The fitted TFT model to be explained.
+        """
+        super().__init__(model)
+
+        if not model._fit_called:
+            raise ValueError("The model needs to be trained before explaining it.")
+
+        self._model = model
+
+    @property
+    def encoder_importance(self):
+        return self._get_importance(
+            weight=self._model.model._encoder_sparse_weights,
+            names=self._model.model.encoder_variables,
+        )
+
+    @property
+    def decoder_importance(self):
+        return self._get_importance(
+            weight=self._model.model._decoder_sparse_weights,
+            names=self._model.model.decoder_variables,
+        )
+
+    def get_variable_selection_weight(self, plot=False) -> Dict[str, pd.DataFrame]:
+        """Returns the variable selection weight of the TFT model.
+
+        Parameters
+        ----------
+        plot
+            Whether to plot the variable selection weight.
+
+        Returns
+        -------
+        TimeSeries
+            The variable selection weight.
+
+        """
+
+        if plot:
+            # plot the encoder and decoder weights
+            self._plot_cov_selection(
+                self.encoder_importance,
+                title="Encoder variable importance",
+            )
+            self._plot_cov_selection(
+                self.decoder_importance,
+                title="Decoder variable importance",
+            )
+
+        return {
+            "encoder_importance": self.encoder_importance,
+            "decoder_importance": self.decoder_importance,
+        }
+
+    def explain(self, **kwargs) -> ExplainabilityResult:
+        """Returns the explainability result of the TFT model.
+
+        The explainability result contains the attention heads of the TFT model.
+        The attention heads determine the contribution of time-varying inputs.
+
+        Parameters
+        ----------
+        kwargs
+            Arguments passed to the `predict` method of the TFT model.
+
+        Returns
+        -------
+        ExplainabilityResult
+            The explainability result containing the attention heads.
+
+        """
+        super().explain()
+        # without the predict call, the weights will still bet set to the last iteration of the forward() method
+        # of the _TFTModule class
+        if "n" not in kwargs:
+            kwargs["n"] = self._model.model.output_chunk_length
+
+        _ = self._model.predict(**kwargs)
+
+        # get the weights and the attention head from the trained model for the prediction
+        attention_heads = (
+            self._model.model._attn_out_weights.squeeze().sum(axis=1).detach()
+        )
+
+        # return the explainer result to be used in other methods
+        return ExplainabilityResult(
+            {
+                0: {
```

Comment:
Is this always relating to horizon 0 only? How about the cases where `predict()` is called with `n > 1` above?

Reply:
I had to set the `0` here to be compatible with the `ForecastingModelExplainer` base class. To get the `attention_heads` the `predict` method of the TFT class has to be called or the attention_heads will not show the correct values. I am not sure why yet. Placing this logic into the explain() method as the `ExplainabilityResult` felt like a sensible choice. 
We could deviate from the `ForecastingModelExplainer` class or add a note to the docstring that the `0` is irrelevant in this context.

Reply:
So if I follow well, here the explanation is for all forecasted horizons at once, right?
I would then propose the following. We can adapt the class `ExplainabilityResult` in order to make it a little bit more flexible:
* It could be used with one explanation per horizon (as now), or
* with one single explanation for all horizons (as required in this case for the TFT).

To accommodate the second case, we could make it possible to build `ExplainabilityResult` with only a `Dict[str, TimeSeries]` (in addition to `Dict[integer, Dict[str, TimeSeries]]`), so we avoid specifying the horizon. We can also adapt `ExplainabilityResult.get_explanation()` to make specifying the `horizon` optional, and not supported if the underlying explanation is not split by horizon.

WDYT? I would find this cleaner than "hacking" the class by using a fake horizon 0.

Reply:
@Cattes any thoughts on this? ^

Reply:
I think its a good idea to change the class to handle the TFT explainer. I didn't want to do it before discussing it. Having the hack with horizon=0 was just to conform with the given api. It was not an intuitive solution. 
I have added `Dict[str, TimeSeries]` to the valid type for class initialization and made the `horizon` optional. 
I also added a few more validations to deal with the different input types explicitly.